modules/storage.js

```diff
@@ -188,4 +178,54 @@ module.exports = class Storage {
   getAstNodesData(uri) {
     return this.astNodesData.get(uri);
   }
+
+  getRuleTempId(ruleId) {
+    return this.ruleTempId.get(ruleId);
+  }
+
+  setRuleTempId(ruleId, ruleTempId) {
+    this.ruleTempId.set(ruleId, ruleTempId);
+  }
+
+  removeRuleTempId(ruleId) {
+    this.ruleTempId.delete(ruleId);
+  }
+
+  setRuleTempIdToTestCase(testCaseStartedId, ruleTempId) {
+    this.ruleTempIdToTestCaseStartedId.set(testCaseStartedId, ruleTempId);
+  }
+
+  getRuleTempIdToTestCase(testCaseStartedId) {
+    return this.ruleTempIdToTestCaseStartedId.get(testCaseStartedId);
+  }
+
+  removeRuleTempIdToTestCase(testCaseStartedId) {
+    this.ruleTempIdToTestCaseStartedId.delete(testCaseStartedId);
+  }
+
+  getStartedChildren(ruleTempId) {
+    return this.startedChildren.get(ruleTempId) || [];
+  }
+
+  setStartedChildren(ruleTempId, child) {
+    this.startedChildren.set(ruleTempId, [
+      ...new Set([...this.getStartedChildren(ruleTempId), child]),
```

Comment:
If you need to hold an unique set of values, you use `Set` directly without transforming it to array.